cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt

```diff
@@ -102,27 +105,60 @@ private class NodeRunner(baseDebugPort: Int, baseMonitoringPort: Int) {
     }
 
     private fun startWindowed(command: List<String>, workingDir: File, nodeName: String): Process {
+
+        val createTempScriptFile = {
+            val tmpFile: File = File.createTempFile("corda_node_script_launcher", "")
+            BufferedWriter(OutputStreamWriter(FileOutputStream(tmpFile))).use {
+                //Start shell to keep window open unless java terminated normally or due to SIGTERM:
+                it.write("""
+                        #!/bin/sh
+                        set -v
+                        unlink $tmpFile
+                        cd "$workingDir"
+                        command -v rlwrap
+                        if [ ${'$'}? -eq 0 ]
```

Comment:
Oook! I think you can jus write `$?` in this _particular_ case because there's no`{` immediately after the `$`.

However, I believe that `$?` is the return code from _the very last_ command executed, which means that **you need to capture its value _immediately_ if you wish to use it again**.
e.g.
```
RC=$?
if [ $RC -eq 0 -o $RC -eq 143 ]
```
I really don't believe that your `$? -eq 0 -o $? -eq 143` lines are doing what you think.

Reply:
OK, this `$?` issue actually looks like a bug in the original script too.